src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -799,15 +794,20 @@ private async Task AcquireWriteLockAsync(CancellationToken cancellationToken)
                         LogExceptions(FlushAsync());
                     }
 
+                    // On cancel, we get TaskCanceledException here instead of OperationCanceledException.
+                    // For consistency, try to ensure we get OperationCanceledException when the cancellationToken has fired.
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
> For consistency, try to ensure we get OperationCanceledException when the cancellationToken has fired.

We have a bunch of places that actually try to do the opposite.
https://github.com/dotnet/corefx/blob/e64cac6dcacf996f98f0b3f75fb7ad0c12f588f7/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/CancellationHelper.cs#L27-L28

I would remove this.

Reply:
Ok, will do. I changed it because some test was explicitly checking for OCE; I'll fix the test.